clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -339,7 +339,7 @@ private boolean maybeSendTransactionalRequest(long now) {
 
                 if (targetNode != null) {
                     if (nextRequestHandler.isRetry())
-                        time.sleep(retryBackoffMs);
+                        time.sleep(Math.min(retryBackoffMs, nextRequestHandler.retryBackoffMs()));
```

Comment:
Hmm, I think it would be better if we just used `nextRequestHandler.retryBackoffMs`. If we want to do the `min` logic, we can do it while doing the override in that class, right? That way we can keep the logic in one place.